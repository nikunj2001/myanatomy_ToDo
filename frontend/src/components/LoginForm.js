import React, { useEffect, useState } from 'react'
import { useDispatch, useSelector } from "react-redux"
import { userLogin } from '../store/asyncMethods/AuthMethods';
import { Toaster, toast } from "react-hot-toast"
import Store from '../store/'

const LoginForm = () => {
    console.log();
    const [state, setState] = useState({ email: '', password: '' });
    const dispatch = useDispatch();
    const { loginErrors } = Store.getState().AuthReducer;
    const handleChange = e => {
        setState({ ...state, [e.target.name]: e.target.value });
    }
    const handleSubmit = e => {
        e.preventDefault();
        Store.dispatch(userLogin(state));
    }
    useEffect(() => {
        if (loginErrors.length > 0) {
            loginErrors.map(err => toast.error(err.msg))
            dispatch({ type: "CLOSE_LOGIN_ERRORS" })
        }
    }, [loginErrors]);
    return (
        <div className='form' >
            <h1>Login Form</h1>
            <Toaster
                position="top-center"
                reverseOrder={false}
            />
            <form onSubmit={handleSubmit}>
                <input type="text" placeholder='Email' name='email' onChange={handleChange} />
                <input type="password" placeholder='Password' name='password' onChange={handleChange} />
                <input type="submit" placeholder='Login' />
            </form>
        </div>
    )
}

export default LoginForm