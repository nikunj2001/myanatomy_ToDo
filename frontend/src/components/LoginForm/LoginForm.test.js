import React from 'react';
import { shallow, configure } from "enzyme";
import Enzyme from 'enzyme';
import Adapter from "@wojtekmaj/enzyme-adapter-react-17"
import LoginForm from "./index"

Enzyme.configure({ adapter: new Adapter() })


describe("LoginForm", () => {
    it("should render my component", () => {
        const wrapper = shallow(<LoginForm />);
        const tree = wrapper.debug();
        expect(tree).toMatchSnapshot();
    });
});