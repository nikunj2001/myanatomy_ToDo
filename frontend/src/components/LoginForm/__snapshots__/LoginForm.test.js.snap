// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateTask should render my component 1`] = `
"<div className=\\"form\\">
  <h1>
    Login Form
  </h1>
  <Toaster position=\\"top-center\\" reverseOrder={false} />
  <form onSubmit={[Function: handleSubmit]}>
    <input type=\\"text\\" placeholder=\\"Email\\" name=\\"email\\" onChange={[Function: handleChange]} />
    <input type=\\"password\\" placeholder=\\"Password\\" name=\\"password\\" onChange={[Function: handleChange]} />
    <input type=\\"submit\\" placeholder=\\"Login\\" />
  </form>
</div>"
`;

exports[`LoginForm should render my component 1`] = `
"<div className=\\"form\\">
  <h1>
    Login Form
  </h1>
  <Toaster position=\\"top-center\\" reverseOrder={false} />
  <form onSubmit={[Function: handleSubmit]}>
    <input type=\\"text\\" placeholder=\\"Email\\" name=\\"email\\" onChange={[Function: handleChange]} />
    <input type=\\"password\\" placeholder=\\"Password\\" name=\\"password\\" onChange={[Function: handleChange]} />
    <input type=\\"submit\\" placeholder=\\"Login\\" />
  </form>
</div>"
`;
